### ISSUE TITLE
Fix list duplication with instantiated objects

### ISSUE THREAD

**Issue by TheSniperFan** [TheSniperFan/unityserializer-ng#16]
When storing lists of instantiated objects (like GameObject references) they get duplicated upon restoring.

**Comment by TheSniperFan** [comment]
__Highly experimental__ fix:

```C#
// LevelLoader.cs, Line 455

// Change
UnitySerializer.RunDeferredActions(process, 2, false);
// To
UnitySerializer.RunDeferredActions(process, 0, false);
```

Again, this is __highly experimental__. While it is plausible that this was merely a typo of the original developer (if he typed on the numerical keypad), it could also cause massive problems elsewhere.

__Highly experimental__, could __eat your babies!__

**Comment by PentagramPro** [comment]
Have same issue. This line:
    // LevelLoader.cs, Line 455
    // Change
    UnitySerializer.RunDeferredActions(process, 2, false);
    // To
    UnitySerializer.RunDeferredActions(process, 0, false);
You can comment it out, because with 0 as a second parameter this method does nothing (except locking and unlocking a mutex). I dont know the consequences of this fix.

All this is very strange. Some lines later you can see another call to this method:
    UnitySerializer.RunDeferredActions(process);

And another, commented out, at line 466:
    //UnitySerializer.RunDeferredActions(2);

Looks like developer experimented with multithreading and just forgot to comment out some debug stuff.

**Comment by TheSniperFan** [comment]
The lack of proper comments has been giving me headaches from the beginning.
I cannot safely remove this line, because it would cause problems elsewhere, if we assume that it wasn't placed there for no reason.

I didn't have the time to make really figure out what changes, if you remove it.

**Comment by TheSniperFan** [comment]
Okay, I checked what exactly it does again. It really seems to be doing nothing.

Fixed in 138361dd74286529d07efa9b9981083ca021637e
